DISPATCH_SUBSCRIBE.md

```diff
@@ -0,0 +1,98 @@
+# Forensics on RDX usage
+
+This is a "working" guess on how RDX works, and how to implement it into your own system.
+
+- working - because it's enough code to actually prove that it functions as advertised
+- guess - in that "Who knows if this is the right way, or not"
+
+## What is the source of this document?
+
+This document is taken entirely from examining the code from the March 2020 version of rdx-demo, and trying to replicate what that does within the firebase sign out code
+
+## What is the high level objective?
+
+- one WC would be able to pass a value into the rdx state model
+- the rdx state model would then be able to do some pure function with that passed value, in our case, square the number
+- any other WC would be able to subscribe to the results of that pure function via a property
+
+## Now, describe the above in code:
+
+This is the simple part, not more difficult than coding in events or properties-down or any other state handling mechanism. Just different semantics/syntax
+
+- within calling WC `dispatch.myModel.takeValue(value)`
+- within the pure function `value = value * value`
+- within the receiving WC's mapState() `myval: MyModelSelectors.squaredValue(state)`
+
+## memoization, under the hood?
+
+- a package called "reselect" provides a memoization layer for us.
+- this doc does not even consider the possibility of bypassing this layer - although some may ask if this is advantageous or possible in some situations
```

Comment:
Reselect does two things:

1. It provides a "public api" to your state, possibly insulating your component mappings from refactoring of the model internals
2. It memoizes projections so any expensive combining of data is repeatedly re-calculated unless the values the result depends on changes